src/datasources/data-frame/components/v2/DataFrameQueryEditorV2.tsx

```diff
@@ -132,21 +132,27 @@ export const DataFrameQueryEditorV2: React.FC<Props> = ({ query, onChange, onRun
     const onDataTableFilterChange = (event?: Event | React.FormEvent<Element>) => {
         if (event) {
             const dataTableFilter = (event as CustomEvent).detail.linq;
-            handleQueryChange({ ...migratedQuery, dataTableFilter });
+            if (dataTableFilter !== migratedQuery.dataTableFilter) {
+                handleQueryChange({ ...migratedQuery, dataTableFilter });
+            }
         }
     };
```

Comment:
```suggestion
    const onDataTableFilterChange = (event?: Event | React.FormEvent<Element>) => {
        if (!event) {
            return;
        }

        const dataTableFilter = (event as CustomEvent).detail.linq;
        if (dataTableFilter === migratedQuery.dataTableFilter) {
            return;
        }

        handleQueryChange({ ...migratedQuery, dataTableFilter });
    };
```

Can we follow this pattern?